tests/Doctrine/Tests/DBAL/Platforms/AbstractPostgreSqlPlatformTestCase.php

```diff
@@ -919,4 +919,19 @@ public function testQuotesDatabaseNameInCloseActiveDatabaseConnectionsSQL()
             true
         );
     }
+
+    public function testGetDefaultValueDeclarationSQLIsQuotedWithLiteral()
+    {
+        $field = [
+            'type' => Type::getType('string'),
+            'default' => "'O'Connor said: \"Hello\" \ \r'"
+        ];
+
+        self::assertSame(sprintf(
+            ' DEFAULT %s',
+            $this->_platform->quoteStringLiteral("'O'Connor said: \"Hello\" \ \r'")
```

Comment:
Doesn't it just duplicate the implementation of `getDefaultValueDeclarationSQL()`? In this case, if the implementation if that method has a bug, this test still will pass. If you only want to make sure that the method was invoked, it should be mocked. But for me, when it comes to testing DB-specific logic, a functional test is much more valuable. I'd remove this in favor of https://github.com/doctrine/dbal/pull/2825/files#r144380548.

Reply:
Great explanation. Was thinking of this too when writing ;) Thanks

Reply:
Following your advice, I removed both tests.